### PR TITLE
Add Developer Guide

### PR DESCRIPTION
**Pull request by lexaknyazev**

### CODE REVIEW

**Review by donmccurdy — APPROVED** [review]
Thanks, I think this addresses my comments. Fine with me to merge when ready, and we'll fill in the placeholder sections afterward.

File: DeveloperGuide.md [hunk 1]
```diff
@@ -0,0 +1,408 @@
+# glTF Universal Textures Developer Guide
```
Comment (donmccurdy) [inline DeveloperGuide.md:1]: Is this still the right title, if we are moving away from the term "Universal Textures"? Suggestion:

```suggestion
# KTX 2.0 / Basis Universal Textures in glTF — Developer Guide
```

Comment (lexaknyazev) [reply DeveloperGuide.md:1]: Good catch. By the way, this document doesn't mention glTF except in the title.

Comment (donmccurdy) [reply DeveloperGuide.md:1]: The KTX section (https://github.com/KhronosGroup/3D-Formats-Guidelines/pull/2) would add another minor reference to glTF. I think that's fine, and we don't need more than that except perhaps a link to `KHR_texture_basisu`. Or would you prefer to remove these references?

Comment (lexaknyazev) [reply DeveloperGuide.md:1]: This guide does not depend on glTF and can be used on its own. Still, there's a value in referring to the glTF extension that enables the integration.

So let's remove the reference from the title and add a reference to the extension.

Comment (donmccurdy) [reply DeveloperGuide.md:1]: Sounds good to me.

**Comment by donmccurdy** [comment]
Seems like we are down to wording tweaks at this point. Let's merge this and make any further changes in additional PRs.

**Pull request merged**